modules/cloudevent-pull-trigger/main.tf

```diff
@@ -0,0 +1,90 @@
+resource "random_string" "suffix" {
+  length  = 4
+  upper   = false
+  special = false
+}
+
+// A dedicated service account for this subscription.
+resource "google_service_account" "this" {
+  project = var.project_id
+
+  account_id   = "${var.name}-${random_string.suffix.result}"
+  display_name = "Delivery account for ${var.private-service.name} in ${var.private-service.region}."
+}
+
+// Lookup the identity of the pubsub service agent.
+resource "google_project_service_identity" "pubsub" {
+  provider = google-beta
+  project  = var.project_id
+  service  = "pubsub.googleapis.com"
+}
+
+locals {
+  // See https://cloud.google.com/pubsub/docs/subscription-message-filter#filtering_syntax
+  filter-elements = concat(
+    [for key, value in var.filter : "attributes.ce-${key}=\"${value}\""],
+    [for key, value in var.filter_prefix : "hasPrefix(attributes.ce-${key}, \"${value}\")"],
+    [for key in var.filter_has_attributes : "attributes:ce-${key}"],
+    [for key in var.filter_not_has_attributes : "NOT attributes:ce-${key}"],
+  )
+}
```

Comment:
I wonder if we should create an internal module to share this logic so that things don't diverge 🤔